include/mbgl/style/conversion_impl.hpp

```diff
@@ -89,29 +89,18 @@ class Convertible {
         new (static_cast<void*>(&storage)) std::decay_t<T>(std::forward<T>(value));
     }
 
-    Convertible(Convertible&& v)
-        : vtable(v.vtable)
-    {
-        if (vtable) {
-            vtable->move(std::move(v.storage), this->storage);
-        }
+    Convertible(Convertible&& v) : vtable(v.vtable) {
+        vtable->move(std::move(v.storage), storage);
     }
 
     ~Convertible() {
-        if (vtable) {
-            vtable->destroy(storage);
-        }
+        vtable->destroy(storage);
     }
 
     Convertible& operator=(Convertible&& v) {
```

Comment:
check `if (this != &other)`  ? otherwise at `Convertible a; a = std::move(a)` storage will be destroyed

Reply:
:+1: done